core/src/test/scala/com/nvidia/spark/rapids/tool/util/ToolUtilsSuite.scala

```diff
@@ -131,4 +136,59 @@ class ToolUtilsSuite extends FunSuite with Logging {
     lazy val hadoopConf = RapidsToolsConfUtil.newHadoopConf()
     hadoopConf.get("property.key1") shouldBe "value1"
   }
+
+  test("parse timeFormat 'HH:MM:SS.FFF' to Long") {
+    val currCalendar = Calendar.getInstance()
+    val acceptedTimeFormat = new SimpleDateFormat("HH:mm:ss.SSS")
+
+    val hour = currCalendar.get(Calendar.HOUR_OF_DAY)
+    val minute = currCalendar.get(Calendar.MINUTE)
+    val seconds = currCalendar.get(Calendar.SECOND)
+    val millis = currCalendar.get(Calendar.MILLISECOND)
+    val currTimeAsStr = acceptedTimeFormat.format(currCalendar.getTime)
+    val duration =
+      hour.hours.toMillis + minute.minutes.toMillis + seconds.seconds.toMillis + millis
+    // test successful parsing
+    StringUtils.parseFromDurationToLongOption(currTimeAsStr).get shouldBe duration
```

Comment:
Curious how this code makes sure that `currTimeAsStr==duration` always?

Reply:
That's a good question:

1. `SimpleDateFormat("HH:mm:ss.SSS")` returns the 24-hour clock in the range [0-23]. Unlike `hh:mm:ss.SSS` which returns the hour in the range from [0-12] 
2. `val hour = currCalendar.get(Calendar.HOUR_OF_DAY)` returns the hour part of the calendar instance in the range [0-24] which will avoid any inconsistent between the two output.
3. Finally, I construct the string using the time components hour, minutes, seconds, milliseconds.